src/main/scala/akka/rabbitmq/ChannelActor.scala

```diff
@@ -131,48 +137,49 @@ class ChannelActor(setupChannel: (Channel, ActorRef) => Any)
           goto(Disconnected) using InMemory()
       }
   }
+
   whenUnhandled {
     case Event(GetState, _) =>
       sender ! stateName
       stay
   }
+
   onTransition {
     case Disconnected -> Connected => log.info("{} connected", self.path)
     case Connected -> Disconnected => log.warning("{} disconnected", self.path)
   }
+
   onTermination {
     case StopEvent(_, Connected, Connected(channel)) =>
       log.debug("{} closing channel {}", self.path, channel)
-      closeIfOpen(channel)
+      close(channel)
   }
+
   initialize()
 
-  def setup(channel: Channel): Channel = {
+  private def setup(channel: Channel): Channel = {
     log.debug("{} setting up new channel {}", self.path, channel)
     channel.addShutdownListener(this)
     setupChannel(channel, self)
     channel
   }
 
-  def dropChannelAndRequestNewChannel(broken: Channel) {
-    log.debug("{} closing broken channel {}", self.path, broken)
-    closeIfOpen(broken)
+  private def dropChannelAndRequestNewChannel(broken: Channel) {
+    dropChannel(broken)
     askForChannel()
   }
 
-  def askForChannel() {
+  private def dropChannel(broken: Channel): Unit = {
+    log.debug("{} closing broken channel {}", self.path, broken)
+    close(broken)
+  }
+
+  private def askForChannel() {
     log.debug("{} asking for new channel", self.path)
     connectionActor ! ProvideChannel
   }
 
-  def connectionActor = context.parent
-
-  def shutdownSignalAppliesToChannel(shutdownSignal: ShutdownSignal, channel: Channel): Boolean =
-    shutdownSignal match {
-      case ParentShutdownSignal           => true
-
-      case amqpSignal: AmqpShutdownSignal => amqpSignal.appliesTo(channel)
-    }
+  private[rabbitmq] def connectionActor = context.parent
 
   @scala.throws[Exception](classOf[Exception])
   override def postRestart(reason: Throwable) {
```

Comment:
@reinierl I see this was a fix for https://github.com/NewMotion/akka-rabbitmq/issues/23. If the ChannelActor indeed somehow restarts, then asking for a new channel (without the old one properly closed) might lead to resource leaks, i.e: multiple channels per ChannelActor.